cva6/tests/custom/sv32/vm_misaligned_superpage.S

```diff
@@ -0,0 +1,207 @@
+
+    #ifdef smode
+      SMODE_ECALL                                                           # changes mode M to S and set the MEPC
+    #else
+     UMODE_ECALL                                                            # changes mode M to U and set the MEPC
+    #endif
+
+# ----------------------Execute test prolog------------------------------
+
+    TEST_PROLOG(check_execute,CAUSE_FETCH_PAGE_FAULT)                       # load the addr and expected cause 
+
+# -------------LEVEL 1 PTE Setup for execute test------------------------
+                                                                            # Setup a new PTE to test execute
+    la a1,check_execute                                                     # loads the address of label vm_en
+    mv a0, a1                                                               # set the VA to PA (identity mapping)
+    ori a2, x0, ( PTE_D | PTE_A | SET_PTE_U | PTE_X | PTE_V)                # sets the permission bits
+    PTE_SETUP_RV32(a1, a2, t1, a0, pgtb_l1, LEVEL1)                         # setup the PTE for level1
+
+# ----------------Set the SATP and change the mode---------------------
+
+    SATP_SETUP_SV32(pgtb_l1)                                                # set the SATP for virtualization
+    la a1,check_execute                                                     # loads the address of check_execute
+    #ifdef smode
+      CHANGE_T0_S_MODE(a1)                                                  # changes mode M to S and set the MEPC
+    #else
+      CHANGE_T0_U_MODE(a1)                                                  # changes mode M to U and set the MEPC
+    #endif
+
+.align 12                                                                   # to make the address misliagned to test execute access
+check_execute:
+
+    li t1, 0x45                                                             # page fault should raise 
+    TEST_STATUS                                                             # checks the status of the test
+
+    #ifdef smode
+      SMODE_ECALL                                                           # changes mode M to S and set the MEPC
+    #else
+     UMODE_ECALL                                                            # changes mode M to U and set the MEPC
+    #endif
+
+test_pass:
+
+    li x1, 0                                                                # Write 0 in x1 if test pass
+    j exit                                                                  # Jump to exit
+
+test_fail:
+
+    li x1, 1                                                                # Write 1 in x1 if test failed
+    j exit                                                                  # Jump to exit
+                                                                   
+trap_handler:
+
+    csrr t0, mcause                                                         # read the value of mcause 
```

Comment:
```suggestion
```

Reply:
No further updates are needed for trap_handler based on our previous discussion